openquake/risklib/scientific.py

```diff
@@ -1517,22 +1518,26 @@ def __init__(self, conditional_loss_poes, return_periods, loss_dt,
                 - risk_investigation_time / return_periods)
 
     # used in post_risk
-    def build_curve(self, years, losses, rlzi=0):
+    def build_curve(self, years, kind, losses, agg_types, rlzi=0):
         """
         Compute EP curves. If years is not None, also AEP and OEP curves.
         """
+        if kind == 'losses':  # for consequences
+            kind = 'loss'
         periods = self.return_periods
         ne = self.num_events[rlzi]
-        dic = {"ep": losses_by_period(losses, periods, ne, self.eff_time)}
-        if len(years):
-            dframe = pandas.DataFrame(dict(year=years, loss=losses))
-            oep = []
-            aep = []
-            for year, df in dframe.groupby('year'):
-                oep.append(df.loss.max())
-                aep.append(df.loss.sum())
-            dic['aep'] = losses_by_period(aep, periods, ne, self.eff_time)
-            dic['oep'] = losses_by_period(oep, periods, ne, self.eff_time)
+        dic = {kind: losses_by_period(losses, periods, ne, self.eff_time)}
+        # NOTE: assuming 'ep' mandatory and 'oep' and 'aep' optional
+        if len(years) and kind == 'loss':
```

Comment:
Perhaps it would be better to change this into
`if len(years) > 1 and kind == 'loss'`
in order to avoid computing `loss_oep` and `loss_aep` when they are equal to `loss`

Reply:
No, it would be confusing to the user because by changing the effective time sometime she would get sometimes some curves  and sometimes not.

Reply:
Ok, so by default all curves will be computed, unless otherwise specified in the ini.